### PR TITLE
Add contributing-guidelines document

### PR DESCRIPTION
**Pull request by GitGab19**
This PR adds a CONTRIBUTING.md file, so that new contributors have a document containing major guidelines to follow.
I'm looking for feedbacks and suggestions!

To have a better readability, you can review it here: https://github.com/GitGab19/stratum/blob/add-contributing-document/CONTRIBUTING.md

### CODE REVIEW

**Review by plebhash — APPROVED** [review]

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  - [Project Structure](#project-structure)
+  - [Contribution workflow](#contribution-workflow)
+  - [Your First Code Contribution](#your-first-code-contribution)
+  
+
+## I Have a Question
+
+> If you want to ask a question, we assume that you have read the documentation available at [stratumprotocol.org/docs](https://stratumprotocol.org).
+
+Before you ask a question, it is best to search for existing [Issues](https://github.com/stratum-mining/stratum/issues) that might help you. In case you have found a suitable issue and still need clarification, you can write your question in this issue.
```
Comment (pavlenex) [inline CONTRIBUTING.md:31]: > you can write your question in this issue.

Not clear, which issuue? I think issues are tasks, for questions we can tell people to ask it in Discord instead?
```suggestion
Best way to ask a question is to hop onto our community [Discord](https://discord.com/invite/fsEW23wFYs). Two most suitable places to post a question are:
- #newbies-qs and
- #dev, for technical questions, suitable for developers building on top of SRI, or contributing to it.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+If you then still feel the need to ask a question and need clarification, we recommend the following:
+
+- Open an [Issue](https://github.com/stratum-mining/stratum/issues/new).
+- Provide as much context as you can about what you're running into.
+  
+OR / AND
+
+- Enter our [Discord](https://discord.gg/fsEW23wFYs) server.
+- Find the most appropriate channel for your question (typically it would be [#newbies-qs](https://discord.com/channels/950687892169195530/1012371967992672321)).
+- Provide as much context as you can and submit your question there.
```
Comment (pavlenex) [inline CONTRIBUTING.md:42]:
```suggestion
- For higher chances of getting a quick reply to your questions, it's recommended to provide as much details as possible about the topic you're trying to learn more on.
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+We will then take care of the issue and answer to your question as soon as possible.
+
+## What Should I Know Before I Get Started
+
+### Important Resources About SRI
+
+In order to have a better overview about what SRI covers, it's really suggested to have a look at the following resources before getting started with contributions.
```
Comment (pavlenex) [inline CONTRIBUTING.md:50]:
```suggestion
In order to have a better overview about what SRI covers, have a look at the following resources before getting started with contributions.
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+The SRI project follows an open contributor model, where anyone is welcome to contribute through reviews, documentation, testing, and patches. Follow these steps to contribute:
+
+1. **Fork the Repository**
+
+2. **Create a Branch** 
+
+3. **Commit Your Changes** 
+
+4. **Submit a Pull Request:** once you're satisfied with your changes, submit a pull request to the original SRI repository. Provide a clear and concise description of the changes you've made. If your pull request addresses an existing issue, reference the issue number in the description. In order to contribute to the protocol implementation, every PR must be opened against `dev` branch. To better understand which is the SRI branches structure, please have a look [here](https://github.com/stratum-mining/stratum/blob/main/RELEASE.md#principal-branches).
```
Comment (Fi3) [inline CONTRIBUTING.md:91]: maybe we should add something about running tests + clippy + fmt before open the PR. We could also adapt the little thing that I wrote about debugging MG tests here

**Review by pavlenex — APPROVED** [review]
I think there may be few more bits and pieces to iron this one out, but since we want to have this asap, imo it's good to go, and we can submit PR's later to iron it out. 

One great page for onboarding is made by @gbks https://bitcoin.design/join/ and we can strive to have something as friendly as that as we move forward, but for now, this is great starting point , Go Gabbo!

File: CONTRIBUTING.md (hunk 4 above)
Comment (GitGab19) [reply CONTRIBUTING.md]: Done here 5544830909ab3d489db23815d9c5e91d6574b5e1

**Pull request merged**